--- qpycore_post_init.cpp-pp    2020-11-16 19:00:18.131242550 +0100
+++ qpycore_post_init.cpp       2020-11-16 19:00:37.408280217 +0100
@@ -153,6 +153,6 @@
     PyQtSlotProxy::mutex = new QMutex(QMutex::Recursive);
 
     // Load the embedded qt.conf file if there is a bundled copy of Qt.
-    if (!qpycore_qt_conf())
-        Py_FatalError("PyQt5.QtCore: Unable to embed qt.conf");
+    //if (!qpycore_qt_conf())
+    //    Py_FatalError("PyQt5.QtCore: Unable to embed qt.conf");
 }
