diff --git a/configure.py b/configure.py
index 131fab7..ae9fb0f 100644
--- a/configure.py
+++ b/configure.py
@@ -2168,7 +2168,7 @@ def run_make(target_config, verbose, exe, makefile_name):
         args.append(makefile_target)
 
     run_command(' '.join(args), verbose)
-
+    return platform_exe+".wasm"
     return platform_exe if os.access(platform_exe, os.X_OK) else None
 
 
@@ -2452,7 +2452,7 @@ def run_test_program(mname, test, verbose):
     the name of the PyQt module being tested.  test is the name of the test
     executable.  verbose is set if the output is to be displayed.
     """
-
+    return [""]
     out_file = 'cfgtest_' + mname + '.out'
 
     # Create the output file, first making sure it doesn't exist.
diff --git a/qpy/QtGui/qpyopengl_value_array.cpp b/qpy/QtGui/qpyopengl_value_array.cpp
index c0ccebe..b275e3e 100644
--- a/qpy/QtGui/qpyopengl_value_array.cpp
+++ b/qpy/QtGui/qpyopengl_value_array.cpp
@@ -208,11 +208,6 @@ static void *convert_values(Array *array, PyObject *values, GLenum gl_type,
             array_type = GL_FLOAT;
             break;
 
-#if defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
-        case 'd':
-            array_type = GL_DOUBLE;
-            break;
-#endif
 
         default:
             PyErr_Format(PyExc_TypeError, "unsupported buffer type '%s'",
@@ -318,14 +313,6 @@ static void *convert_values(Array *array, PyObject *values, GLenum gl_type,
         element_size = sizeof (GLfloat);
         break;
 
-#if defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
-#if GL_DOUBLE != GL_FLOAT
-    case GL_DOUBLE:
-        convertor = convert_double;
-        element_size = sizeof (GLdouble);
-        break;
-#endif
-#endif
 
     default:
         Py_DECREF(seq);
diff --git a/sip/QtGui/QtGuimod.sip b/sip/QtGui/QtGuimod.sip
index d66546d..a5eeb3c 100644
--- a/sip/QtGui/QtGuimod.sip
+++ b/sip/QtGui/QtGuimod.sip
@@ -87,7 +87,6 @@ WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 %Include qopenglshaderprogram.sip
 %Include qopengltexture.sip
 %Include qopengltextureblitter.sip
-%Include qopengltimerquery.sip
 %Include qopenglversionfunctions.sip
 %Include qopenglvertexarrayobject.sip
 %Include qopenglwindow.sip
diff --git a/sip/QtGui/qguiapplication.sip b/sip/QtGui/qguiapplication.sip
index 01fadd5..e7aa488 100644
--- a/sip/QtGui/qguiapplication.sip
+++ b/sip/QtGui/qguiapplication.sip
@@ -32,7 +32,7 @@ class QGuiApplication : QCoreApplication
         sipTypeDef **type;
         int yes, no;
     } graph[] = {
-    #if QT_VERSION >= 0x050100 && defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
+    #if 0
         {sipName_QOpenGLTimeMonitor, &sipType_QOpenGLTimeMonitor, -1, 1},
     #else
         {0, 0, -1, 1},
@@ -69,7 +69,7 @@ class QGuiApplication : QCoreApplication
         {0, 0, -1, 16},
     #endif
         {sipName_QGuiApplication, &sipType_QGuiApplication, -1, 17},
-    #if QT_VERSION >= 0x050100 && defined(SIP_FEATURE_PyQt_Desktop_OpenGL)
+    #if 0
         {sipName_QOpenGLTimerQuery, &sipType_QOpenGLTimerQuery, -1, 18},
     #else
         {0, 0, -1, 18},
