--- ../QtWidgets-pp/sipQtWidgetsQApplication.cpp	2020-11-15 23:21:37.477364813 +0100
+++ sipQtWidgetsQApplication.cpp	2020-11-17 00:00:53.914131243 +0100
@@ -197,6 +197,15 @@
     char sipPyMethods[8];
 };
 
+#include <QtPlugin>  //Import here as lines above contain the path to the current directory (which will confuse patch)
+// Remeber the last QApplication created by Python and provide a way for pyodide/src/main.c to call ->exec() on it
+static sipQApplication *lastApp;
+
+extern "C" void execLastQApp();
+void execLastQApp() {
+    lastApp->exec();
+};
+
 sipQApplication::sipQApplication(int& a0,char**a1,int a2):  ::QApplication(a0,a1,a2), sipPySelf(SIP_NULLPTR)
 {
     memset(sipPyMethods, 0, sizeof (sipPyMethods));
@@ -1851,6 +1860,11 @@
         int argc;
         char **argv;
         
+        // Force loading of required & statically linke plugins
+        Q_IMPORT_PLUGIN(QMinimalIntegrationPlugin);
+        Q_IMPORT_PLUGIN(QWasmIntegrationPlugin);
+        Q_IMPORT_PLUGIN(QOffscreenIntegrationPlugin);
+        
         // Convert the list.
         if ((argv = pyqt5_qtwidgets_from_argv_list(a0, argc)) == NULL)
             sipIsErr = 1;
@@ -1883,7 +1897,7 @@
             sipCpp->sipPySelf = sipSelf;
 
             sipCallHook("__pyQtQAppHook__");
-
+            lastApp = sipCpp;  // Remember this as the last app created
             return sipCpp;
         }
     }
